yaksh/views.py

```diff
@@ -402,12 +402,28 @@ def start(request, questionpaper_id=None, attempt_num=None):
         return my_render_to_response('yaksh/intro.html', context,
                                      context_instance=ci)
     else:
-        ip = request.META['REMOTE_ADDR']
-        if not hasattr(user, 'profile'):
-            msg = 'You do not have a profile and cannot take the quiz!'
+        attempted_paper = AnswerPaper.objects.filter(user=user,
```

Comment:
A suggestion, you can have a get over here instead of filter because in future there will never be multiple answer papers with same attempt number for a given question paper.
Using try except will help, if get fails with DoesNotExist exception then you can create an answer paper for that attempt number. If MultipleObjectsReturned exception then you can take the last answer paper for that attempt.

Reply:
resolved